File: Day5/Pattern2.py
# 1,2,6,-7,24,-26,120,-121,720,-722,...

def fact(n):
    if n == 1 or n == 0:
        return 1
    while n > 0:
        return n * fact(n-1)
negative = True
count = 0

for i in range(1,7):
    if i > 2:
        print(fact(i),end=' ')
        if count == 0:
            if negative:
                print((fact(i)+1)*-1,end=' ')
            else:
                print(fact(i)+1,end=' ')
            count = 2
        elif count == 2:
            if negative:
                print((fact(i)+2)*-1,end=' ')
            else:
                print(fact(i)+2,end=' ')
            count = 0
    else:
        print(fact(i),end=' ')

File: Day1/Pattern2.java
package PlacementTraining.Day1;

public class Pattern2 {
    public static void main(String[] args) {
        int n = 13;
        int[] arr = pattern(n);
        for (int i = 0; i < arr.length; i++) {
            System.out.println(arr[i]);
        }
    }
    static int[] pattern(int n){
        int size = n - (n / 4);
        int[] arr = new int[size];

        boolean negative = false;

        int count = 1;
        int index = 0;
        while(count <= n){
            if(count % 4 != 0){
                if(!negative){
                    arr[index] = count;
                    negative = true;
                }else{
                    arr[index] = count * -1;
                    negative = false;
                }
                index++;
                count++;
            }else{
                count++;
            }
        }
            


        return arr;
    }
}
